File: backend/app/schemas/scan.py
"""扫码相关模型"""
from datetime import datetime

from pydantic import BaseModel


class ScanCreate(BaseModel):
    barcode: str
    source: str = "manual"  # camera / manual


class AllergenHit(BaseModel):
    allergen_name: str
    ingredient: str


class AnalysisOut(BaseModel):
    score: int
    level: str
    allergen_hits: list[AllergenHit] = []
    recommended: bool


class ScanOut(BaseModel):
    id: int
    barcode: str
    product_id: int | None = None
    product_name: str = ""
    score: int = 0
    level: str = ""
    has_allergen: bool = False
    source: str = "manual"
    created_at: datetime
    analysis: AnalysisOut | None = None

File: frontend/src/api/scan.ts
import request from './request'

export interface AllergenHit {
  allergen_name: string
  ingredient: string
}

export interface Analysis {
  score: number
  level: string
  allergen_hits: AllergenHit[]
  recommended: boolean
}

export interface ScanRecord {
  id: number
  barcode: string
  product_id: number | null
  product_name: string
  score: number
  level: string
  has_allergen: boolean
  source: string
  created_at: string
  analysis: Analysis | null
}

export const createScanApi = (barcode: string, source = 'manual') =>
  request.post<ScanRecord>('/scan', { barcode, source }).then((r) => r.data)

export const getScanApi = (scanId: number) =>
  request.get<ScanRecord>(`/scan/${scanId}`).then((r) => r.data)

export const getScanHistoryApi = () => request.get<ScanRecord[]>('/scan/history').then((r) => r.data)
